symphony/bdk/core/auth/auth_session.py

```diff
@@ -96,4 +102,57 @@ async def session_token(self):
 
     @property
     async def key_manager_token(self):
+        """
+
+        :return: an empty string since there is no key manager token in OBO mode.
+        """
         return ""
+
+
+class AppAuthSession:
+    """Extension application RSA authentication handle to store the tokens. It uses a
+    :class:`symphony.bdk.core.auth.ext_app_authenticator.ExtensionAppAuthenticator` to actually authenticate and
+    retrieve the tokens.
+    """
+    def __init__(self, authenticator, app_token: str):
+        """
+
+        :param authenticator: the :class:`symphony.bdk.core.auth.ext_app_authenticator.ExtensionAppAuthenticator`
+          which will actually perform the authentication
+        :param app_token: the application token
+        """
+        self._authenticator = authenticator
+        self._app_token = app_token
+        self._expire_at = -1
+        self._symphony_token = ""
+
+    async def refresh(self) -> None:
+        """Triggers re-authentication to refresh the tokens.
+
+        :return: None
+        """
+        app_tokens = await self._authenticator.authenticate_and_retrieve_tokens(self._app_token)
+        self._symphony_token = app_tokens.symphony_token
+        self._app_token = app_tokens.app_token
+        self._expire_at = app_tokens.expire_at
+
+    def symphony_token(self) -> str:
```

Comment:
Question mainly for myself :) No need to annotate with `@property` in this case?

Reply:
We could. But IMHO `@property` is really useful when we need getters and setters with specific logic inside. With no setter, added value is less obvious.

Reply:
Done in https://github.com/SymphonyPlatformSolutions/symphony-api-client-python/pull/153/commits/423447eeb207ba23964499cd4d9df8f16708bba8